packages/prettier-plugin-java/src/printers/classes.ts

```diff
@@ -588,15 +592,11 @@ export class ClassesPrettierVisitor extends BaseCstPrettierPrinter {
   receiverParameter(ctx: ReceiverParameterCtx) {
     const annotations = this.mapVisit(ctx.annotation);
     const unannType = this.visit(ctx.unannType);
-    const identifier = ctx.Identifier
-      ? concat([ctx.Identifier[0], ctx.Dot![0]])
-      : "";
 
-    return rejectAndJoin("", [
-      rejectAndJoin(" ", annotations),
+    return rejectAndJoin(" ", [
+      ...annotations,
       unannType,
-      identifier,
-      ctx.This[0]
+      rejectAndConcat([ctx.Identifier?.[0], ctx.Dot?.[0], ctx.This[0]])
     ]);
```

Comment:
There was not any space before and after the unannType, was it an issue ?

Reply:
It was -- previously, the unannType was squashed up against the annotations (if there were any) and the identifier/`this`. It just so happened that we didn't have any tests for this, and I suppose it's such a rare syntax that the issue wasn't encountered elsewhere.